util/test/sub_test.py

```diff
@@ -2311,6 +2322,24 @@ def filter_errors(output_in, pre_exec_output, execgoodfile, execlog):
                                              stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
                         sys.stdout.write(p.communicate()[0])
 
+                    if globalGoodfile:
```

Comment:
Not something I think must be done for this PR but there's a lot of (mostly) duplicate code here when it comes to processing these scripts (goodfile, prediff, etc.)

It'd be nice to factor this out into a general "execute this script and get me the resulting stdout" function.